File: Day3/examples/hello_omp.c

#include <stdio.h>

#if defined(_OPENMP)
#include <omp.h>
#endif

int main(int argc, char **argv)
{
    int nthreads, tid;

    nthreads=1;
    tid=0;

#if defined(_OPENMP)
    nthreads = omp_get_max_threads();
    tid = omp_get_num_threads();
#endif

    printf("Total number of threads: %d    number of active threads: %d\n",
           nthreads, tid);

    nthreads=1;
    tid=0;
#if defined(_OPENMP)
#pragma omp parallel default(none) private(tid) shared(nthreads)
    {
        tid=omp_get_thread_num();
        nthreads=omp_get_num_threads();

        printf("Hello World from thread: %d\n",tid);
        if (tid == 0) {
            printf("Number of active threads: %d\n", nthreads);
        }
    }

#endif
    return 0;
}

File: Day3/examples/openmp/hello_omp.f90

PROGRAM hello
  INTEGER :: nthreads, tid
  INTEGER, EXTERNAL :: omp_get_max_threads, omp_get_num_threads, &
      omp_get_thread_num

  nthreads = 1
  tid = 1

!$  nthreads = omp_get_max_threads();
!$  tid = omp_get_num_threads();

  PRINT*,'Total number of threads: ',nthreads, &
      '  Number of active threads:',tid

  nthreads=1
  tid=0

!$OMP parallel default(none) private(tid) shared(nthreads)    
!$ tid=omp_get_thread_num();
!$ nthreads=omp_get_num_threads();

  PRINT*, 'Hello World from thread: ',tid
  IF (tid == 0) THEN
      PRINT*,'Number of active threads:', nthreads
  END IF
!$OMP end parallel
    
END PROGRAM hello
